### PR TITLE
External Monitoring - Alerts / ApiStatus

### PR DESCRIPTION
**Pull request by julienstroheker**
Add Alerts / Node / ApiStatus check

### CODE REVIEW

File: pkg/monitor/alerts.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		return fmt.Errorf("unexpected status code %d", resp.StatusCode)
+	}
+
+	var alerts []model.Alert
+	err = json.NewDecoder(resp.Body).Decode(&alerts)
+	if err != nil {
+		return err
+	}
+
+	for _, alert := range alerts {
```
Comment (jim-minter) [inline pkg/monitor/alerts.go]: build a `map[alert.Labels["alertname"]]int`, populate it, then walk it emitting gauges (i.e. uniq the gauges)

File: pkg/monitor/const.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package monitor
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+const (
```
Comment (jim-minter) [inline pkg/monitor/const.go]: please unexport everything that doesn't need to be exported

File: pkg/monitor/const.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	// MetricAPIServerHeatlth is the metric name for the api server health
+	MetricAPIServerHeatlth string = "ApiServerCode"
```
Comment (jim-minter) [inline pkg/monitor/const.go]: typo Health

File: pkg/monitor/healthz.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	err := cli.Discovery().RESTClient().
+		Get().
+		Context(ctx).
+		AbsPath("/healthz").
+		Do().
+		StatusCode(&statusCode).
+		Error()
+	if err != nil && statusCode == 0 {
+		return fmt.Errorf("API Server is Unhealthy - %s", err.Error())
+	}
```
Comment (jim-minter) [inline pkg/monitor/healthz.go]: rather than generate a synthetic error, why not return the statusCode and see if it `== http.StatusOK` in the caller?

**Comment by julienstroheker** [comment]
Thanks @jim-minter for the review and the update :)

**Pull request merged**